File: frontend/src/types/health.ts
export type ComponentState = "ok" | "degraded" | "down" | "not_configured";
export type OverallState = Exclude<ComponentState, "not_configured">;

export type ComponentStatus = {
  status: ComponentState;
  detail?: string | null;
};

export type HealthResponse = {
  status: OverallState;
  version: string;
  environment: string;
  timestamp: string;
  python_version: string;
  components: Record<string, ComponentStatus>;
};

File: backend/app/schemas/health.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

ComponentState = Literal["ok", "degraded", "down", "not_configured"]
OverallState = Literal["ok", "degraded", "down"]


class ComponentStatus(BaseModel):
    status: ComponentState
    detail: str | None = None


class HealthResponse(BaseModel):
    status: OverallState
    version: str
    environment: str
    timestamp: datetime
    python_version: str
    components: dict[str, ComponentStatus] = Field(default_factory=dict)
